tests/parametric/test_headers_tracestate_dd.py

```diff
@@ -574,3 +574,97 @@ def test_headers_tracestate_dd_propagate_propagatedtags_change_sampling_reset_dm
+                    ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                    ["tracestate", "dd=s:-1," + other_vendors],
+                ],
+            )
+
+            # 3) Input: 31 list-members without 'dd' in the tracestate string
+            headers3 = make_single_request_and_get_inject_headers(
+                test_library,
+                [
+                    ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                    ["tracestate", other_vendors],
+                ],
+            )
+
+            # 4) Input: No tracestate string
+            headers4 = make_single_request_and_get_inject_headers(
+                test_library, [["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],],
+            )
+
+        # 1) Input: 32 list-members with 'dd' at the end of the tracestate string
+        _, tracestate1 = get_tracecontext(headers1)
+        tracestate1String = str(tracestate1)
+        assert "key31=value31" in tracestate1String
+        assert tracestate1String.startswith("dd=")
+        assert len(tracestate1String.split(",")) == 32
+
+        # 2) Input: 32 list-members with 'dd' at the beginning of the tracestate string
+        _, tracestate2 = get_tracecontext(headers2)
+        tracestate2String = str(tracestate2)
+        assert "key31=value31" in tracestate2String
+        assert tracestate2String.startswith("dd=")
+        assert len(tracestate1String.split(",")) == 32
+
+        # 3) Input: 31 list-members without 'dd' in the tracestate string
+        _, tracestate3 = get_tracecontext(headers3)
+        tracestate3String = str(tracestate3)
+        assert "key31=value31" in tracestate3String
+        assert tracestate3String.startswith("dd=")
+        assert len(tracestate1String.split(",")) == 32
+
+        # 4) Input: No tracestate string
+        _, tracestate4 = get_tracecontext(headers4)
+        tracestate4String = str(tracestate4)
+        assert tracestate4String.startswith("dd=")
+        assert len(tracestate1String.split(",")) == 1
+
+    @temporary_enable_propagationstyle_default()
+    def test_headers_tracestate_dd_evicts_32_or_greater_list_members(self, test_agent, test_library):
+        """
+        harness sends a request with both tracestate and traceparent.
+        the last list-member in the input tracestate is removed from the output
+        tracestate string because the maximum number of list-members is 32.
+        """
+        with test_library:
+            other_vendors = ",".join("key%d=value%d" % (i, i) for i in range(1, 32))
+
+            # 1) Input: 32 list-members without 'dd' in the tracestate string
+            headers1 = make_single_request_and_get_inject_headers(
+                test_library,
+                [
+                    ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                    ["tracestate", other_vendors + ",key32=value32"],
```

Comment:
> This `tracestate` value is [not valid according the W3C standard](https://www.w3.org/TR/trace-context/#tracestate-header-field-values).

Can you clarify what isn't valid about this tracestate? When the tracestate arrives into the application it should have a string looking like:
```
tracestate="key1=value1,key2=value2,...,key31=value31,key32=value32"
```

Since tracestate has 32 entries, it is still valid. However, when we propagate an outgoing tracestate then it would require an additional entry so we evict the last one.

This is the scenario I'm trying to test. Did I implement it incorrectly?

Reply:
> At least, this is the scenario I'm trying to test. Did I implement it incorrectly?

And this completely fine. It was me who misread it, so sorry 🙇 
I thought it was `dd` member + 32 other members in the list.
Your test does what is expected and the Java tracer passes it. It was me who was faulty 😓